src/Microsoft.AspNet.Mvc.Razor/RazorTextWriter.cs

```diff
@@ -101,13 +101,13 @@ public override void Write(string value)
         public override void Write(IHtmlContent value)
         {
             var htmlTextWriter = TargetWriter as HtmlTextWriter;
-            if (htmlTextWriter == null)
+            if (htmlTextWriter != null)
             {
-                value.WriteTo(TargetWriter, HtmlEncoder);
+                htmlTextWriter.Write(value);
             }
             else
             {
-                htmlTextWriter.Write(value);
+                value.WriteTo(TargetWriter, HtmlEncoder);
             }
```

Comment:
As above. Also mentioned in the PR / commit description: "make `null` checks of `HtmlTextWriters` consistent".